From: =?utf-8?q?Nick_=C3=98stergaard?= <oe.nick@gmail.com>
Date: Sun, 18 Dec 2016 22:16:22 +0100
Subject: Fix issue #170 -- for real

(Notes by Carsten Schoenert)
Upstream fix for kicad-doc commited after the release of v4.0.5.
This patch will automatically fall out of the patch queue with the next
version 4.0.6.
This commit isn't fixing the installation of the default images. For
this we need another fix.
---
 doc/CMakeModules/KiCadDocumentation.cmake | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/doc/CMakeModules/KiCadDocumentation.cmake b/doc/CMakeModules/KiCadDocumentation.cmake
index e4c91e46..884f46bf 100644
--- a/doc/CMakeModules/KiCadDocumentation.cmake
+++ b/doc/CMakeModules/KiCadDocumentation.cmake
@@ -136,7 +136,14 @@ macro( KiCadDocumentation DOCNAME )
             add_dependencies( ${DOCNAME} ${DOCNAME}_html_${LANGUAGE} )
 
             install( FILES ${CMAKE_CURRENT_BINARY_DIR}/${LANGUAGE}/${DOCNAME}.html DESTINATION ${KICAD_DOC_PATH}/${LANGUAGE} COMPONENT html-${LANGUAGE} )
-            install( DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/${LANGUAGE}/images/ DESTINATION ${KICAD_DOC_PATH}/${LANGUAGE}/images COMPONENT html-${LANGUAGE} OPTIONAL PATTERN "*.png")
+
+            # It seems to be nescesary to use diffrent install commands to
+            # install the images properly without having other lagunages
+            # polluting the images folder
+            install( DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/${LANGUAGE}/images/${LANGUAGE} DESTINATION ${KICAD_DOC_PATH}/${LANGUAGE}/images COMPONENT html-${LANGUAGE} OPTIONAL PATTERN "*.png")
+            install( DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/${LANGUAGE}/images/icons DESTINATION ${KICAD_DOC_PATH}/${LANGUAGE}/images COMPONENT html-${LANGUAGE} OPTIONAL PATTERN "*.png")
+            file(GLOB UNIVERSAL_IMAGES "${CMAKE_CURRENT_BINARY_DIR}/${LANGUAGE}/images/*.png")
+            install( FILES ${UNIVERSAL_IMAGES} DESTINATION ${KICAD_DOC_PATH}/${LANGUAGE}/images COMPONENT html-${LANGUAGE} )
         endif()
 
 
